syntax = "proto3";

package usersProto;

// The greeting service definition.
service DatabaseService {
  rpc getUserById (Id) returns (User) {}
  rpc registerNewUser (User) returns (RegistrationResult) {}
  rpc enumerateUsers (Pagination) returns(stream User) {}
}

message User {
  int64 id = 1;
  string email = 2;
  string nickname = 3;
  string password = 4;
  bool isActive = 5;
}

message Id {
  int32 id = 1;
}

message RegistrationResult {
  int64 resultId = 1;
}

message Pagination {
  int32 pageSize = 1;
  int32 pageNumber =2;
}
